### PR TITLE
1893 - move glooctl version check to individual commands for non-kube…

### PR DESCRIPTION
**Pull request by bdecoste**
### Solution
Currently when any glooctl command is run there is an initial kube-based check of the client/server version. In non-kube envs (e.g. consul) this causes a significant delay in every command while glooctl times out looking for kube and issues a kube-specific warning. By moving the version check to the individual commands the version check can only be done in kube (e.g. non-consul) environments 

I have only updated the GET command in initial PR pending review,. 

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/1893

### CODE REVIEW

**Comment by bdecoste** [comment]
Request a review of both commits. The first commit would require new, duplicate code to be added to many of the cmds. The second commit instead adds the version check to a function that is already checking to see if consul is used.

File: projects/gloo/cli/pkg/prerun/prerun_consul.go [hunk 1]
```diff
@@ -6,7 +6,11 @@ import (
 	"github.com/solo-io/go-utils/errors"
 )
 
-func EnableConsulClients(consul options.Consul) error {
+func EnableConsulClients(opts *options.Options, consul options.Consul) error {
+	if err := VersionMismatchWarning(opts, consul); err != nil {
+		return err
+	}
```
Comment (GrahamGoudeau) [inline projects/gloo/cli/pkg/prerun/prerun_consul.go]: I think this isn't the right place for the version warning to originate from. Can we keep the change in version_warning.go, but instead of passing in the Consul object, can we just get it from `opts.Route.Consul`? And then revert this change here in prerun_consul.go.

File: projects/gloo/cli/pkg/prerun/version_warning.go [hunk 2]
```diff
@@ -20,8 +19,11 @@ const (
 	ContainerNameToCheck = "discovery"
 )
 
-func VersionMismatchWarning(opts *options.Options, cmd *cobra.Command) error {
-	return WarnOnMismatch(os.Args[0], versioncmd.NewKube(opts.Metadata.Namespace), &defaultLogger{})
+func VersionMismatchWarning(opts *options.Options, consul options.Consul) error {
+	if !consul.UseConsul {
```
Comment (GrahamGoudeau) [inline projects/gloo/cli/pkg/prerun/version_warning.go]: Can we add a comment on this conditional explaining why it's necessary? ie what the effect is if it's not here (the current behavior)

File: projects/gloo/cli/pkg/cmd/add/root.go [hunk 3]
```diff
@@ -18,7 +18,7 @@ func RootCmd(opts *options.Options, optionsFunc ...cliutils.OptionsFunc) *cobra.
 			if err := prerun.CallParentPrerun(cmd, args); err != nil {
```
Comment (GrahamGoudeau) [inline projects/gloo/cli/pkg/cmd/add/root.go]: Reminder that you'll need to add a changelog entry. Let me know if you need help with that

File: projects/gloo/cli/pkg/prerun/version_warning.go [hunk 4]
```diff
@@ -20,8 +19,11 @@ const (
...
+		return WarnOnMismatch(os.Args[0], versioncmd.NewKube(opts.Metadata.Namespace), &defaultLogger{})
+	}
+	return nil
```
Comment (GrahamGoudeau) [inline projects/gloo/cli/pkg/prerun/version_warning.go]: Super nitpick, but would slightly prefer to see the logic framed positively:

```go
if consul.UseConsul {
  return nil
}

return WarnOnMismatch(...)
```

File: projects/gloo/cli/pkg/prerun/prerun_consul.go (hunk 1 above)
Comment (bdecoste) [reply projects/gloo/cli/pkg/prerun/prerun_consul.go]: The problem is that it's not always opts.Route.Consul. It can also be opts.Remove.Consul, opts.Get.Consul. That's why I am passing in opts (needed for the namespace) and the specific Consul based on the cmd.

**Comment by rickducott** [comment]
/kick had to fix the regression cluster due to #1944

**Comment by rickducott** [comment]
There is some existing tech debt where the consul flags seem duplicated and we could clean up the structure, which would affect the implementation here. It is also seemingly an area without much of our typical CLI testing. This change seems innocuous and is blocking an enterprise release, so I am filing a follow up issue.

**Comment by rickducott** [comment]
Chatted with Graham and dismissed review.
